quinn/examples/connection.rs

```diff
@@ -0,0 +1,62 @@
+//! This example intends to use the smallest amout of code to make a simple QUIC connection.
+//!
+//! The server issues it's own certificate and passes it to the client to trust.
+//!
+//! Run:
+//! ```
+//! $ cargo run --example connection
+//! ```
+//!
+//! This example will make a QUIC connection on localhost and you should see smth like on stdout:
+//! ```
+//! [server] incoming connection: id=3680c7d3b3ebd250 addr=127.0.0.1:50469
+//! [client] connected: id=61a2df1548935aeb, addr=127.0.0.1:5000
+//! ```
+
+use failure::Error;
+use futures::{future, Future, Stream};
+use tokio::runtime::current_thread::Runtime;
+
+mod common;
+use common::{ipv4_addr, make_client_endpoint, make_server_endpoint};
+
+const SERVER_PORT: u16 = 5000;
+
+fn main() -> Result<(), Error> {
+    let mut runtime = Runtime::new()?;
+
+    let (driver, incoming, server_cert) =
+        make_server_endpoint(&ipv4_addr(0, 0, 0, 0, SERVER_PORT))?;
+    // drive UDP socket
+    runtime.spawn(driver.map_err(|e| eprintln!("IO error: {}", e)));
```

Comment:
Because a driver failure is fatal to the associated endpoint, I think it would be better to panic here (e.g. `panic!("IO error: {}", e);`). We don't want people following these examples and getting confused when a poorly configured server hangs doing nothing because they missed a single line on stderr.